cmd/vcfToFa/vcfToFa.go

```diff
@@ -0,0 +1,73 @@
+// Command Group: "VCF Tools"
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/vcf"
+	"log"
+)
+
+func vcfToFa(inVcfFilename string, inFaFilename string, outFaFilename string, useAlt bool) {
+	var seqsOrdered []fasta.Fasta
+	var seqsLookup fasta.FastaMap
+	var vcfRecords <-chan vcf.Vcf
+
+	// We hang on to the ordered version of the fasta sequences so that we could
+	// output the edited sequences (those in the map) in the same order that
+	// we originally read them, but we do not have that currently
+	seqsOrdered = fasta.Read(inFaFilename)
+	seqsLookup = fasta.ToMap(seqsOrdered)
+
+	// we could do some sort of header check here to make sure the fa matches the vcf
+	vcfRecords, _ = vcf.GoReadToChan(inVcfFilename)
+	for v := range vcfRecords {
+		if !(vcf.IsBiallelic(v) && vcf.IsSubstitution(v)) {
+			log.Fatal("Error: currently we only handle biallelic substitutions\n")
+		}
+
+		// check to make sure the fa matches the vcf before we edit
+		if seqsLookup[v.Chr][v.Pos-1] != dna.StringToBase(v.Ref) {
+			log.Fatal("Error: base in fasta didn't match ref base from VCF record\n")
+		}
+
+		if useAlt {
+			seqsLookup[v.Chr][v.Pos-1] = dna.StringToBase(v.Alt[0])
+		}
+	}
+
+	fasta.WriteMap(outFaFilename, seqsLookup)
```

Comment:
The fasta map just has pointers to the sequences in the slice so you could just use `fasta.Write(seqsOrdered)` and get the properly ordered output unless there is a specific reason you want to write from the map.

Reply:
So we can still take advantage of map lookup while doing normal slice based writing! Such a solution

Reply:
Yes, you are correct.  I had thought about doing that, but I was thinking that in the future people might expand to insertions, which will then have the potential to change where the DNA is stored in memory, if the space needs to be expanded.  I could leave this for that future person to figure out by putting a comment.  Another option would be to add an (optional) ordering to the WriteMap function, or some function that "updates" the slice by looping through slice, hitting the map, and setting the seq of the fasta equal to the seq of the map.  Do people have a preference?

Reply:
Good point. 

Personally I prefer the update option.